src/erlfmt_format.erl

```diff
@@ -528,7 +528,24 @@ fun_to_algebra({type, Meta, Args, Result}) ->
         group(break(nest(Body, ?INDENT, break), <<"">>, <<")">>))
     end).
 
-clauses_to_algebra([Clause | _] = Clauses) ->
+clauses_to_algebra([Clause]) when is_tuple(Clause) ->
+    HasBreak = clause_has_break(Clause),
+    case HasBreak of
+        true -> clauses_to_algebra_([Clause]);
+        false ->
+            %% This is a special case for a single clause,
+            %% with none of its own breaks.
+            %% We want to keep it in a single line,
+            %% even if it has comments that would usually force breaks using:
+            %% `group(Doc)`
+            Meta = element(2, Clause),
+            Doc = do_expr_to_algebra(Clause),
+            combine_comments(Meta, maybe_wrap_in_parens(Meta, group(Doc)))
```

Comment:
Will this work correctly for something like following? 
```erl
receive
    1 -> one; % one
    2 -> two % two
end.
```

This also shouldn't break, I think it should be formatted as:
```erl
receive
    % one
    1 -> one;
    % two
    2 -> two
end.
```

Reply:
Both are formatted to
```erlang
receive
    % one
    1 ->
        one;
    % two
    2 ->
        two
end.
```

Reply:
The intention was only to handle the single clause case, I could extend it to handle more than one clause, if you think that would be better.

Reply:
Yes, I think we should. This wasn't clear in the issue I created, but I intended for it to work like that for multiple clauses as well. I think we should merge this as is for now, keep the issue open, and solve the multi-clause case next.

Reply:
Sounds good, I updated the description not to say "fixes".